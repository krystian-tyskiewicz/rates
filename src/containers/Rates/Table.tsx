import Table from '@mui/material/Table'
import TableBody from '@mui/material/TableBody'
import TableCell from '@mui/material/TableCell'
import TableHead from '@mui/material/TableHead'
import TableRow from '@mui/material/TableRow'
import Paper from '@mui/material/Paper'

import { Rate } from 'types'

type Props = {
  rates: Rate[]
}

const RatesTable = ({ rates }: Props) => {
  return (
    <Paper sx={{ overflowX: 'auto' }}>
      <Table>
        <TableHead>
          <TableRow>
            <TableCell>Currency</TableCell>
            <TableCell>Code</TableCell>
            <TableCell>Mid</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {rates.map((rate) => (
            <TableRow key={rate.code}>
              <TableCell>{rate.currency}</TableCell>
              <TableCell>{rate.code}</TableCell>
              <TableCell>{rate.mid}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </Paper>
  )
}

export default RatesTable
