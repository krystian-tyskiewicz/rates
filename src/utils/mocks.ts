export const rates = [
  {
    currency: 'ariary (Madagaskar)',
    code: 'MGA',
    mid: 0.000885,
  },
  {
    currency: 'balboa (Panama)',
    code: 'PAB',
    mid: 4.0347,
  },
]
