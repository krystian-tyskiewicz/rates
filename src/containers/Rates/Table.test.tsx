import { render } from '@testing-library/react'

import Table from './Table'
import { rates } from 'utils/mocks'

it('renders correctly', () => {
  const { container } = render(<Table rates={rates} />)
  expect(container).toMatchSnapshot()
})
