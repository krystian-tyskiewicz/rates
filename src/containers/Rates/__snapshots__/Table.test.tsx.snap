// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1755g4c-MuiPaper-root"
  >
    <table
      class="MuiTable-root css-rqglhn-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            Currency
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            Code
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            Mid
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            ariary (Madagaskar)
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            MGA
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            0.000885
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            balboa (Panama)
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            PAB
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            4.0347
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
